Update intra_inter_benchmark

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22051 Update intra_inter_benchmark**
* #22047 Experimental option to use single thread pool
* #22045 Limit overall number of threads used by TBB

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Differential Revision: [D15933951](https://our.internmc.facebook.com/intern/diff/D15933951)

Commit:
Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

cpuhrsch commented on binaries/intra_inter_benchmark.cc:
```diff
@@ -70,36 +113,49 @@ int main(int argc, char** argv) {
+            << ", task type: " << FLAGS_task_type << std::endl;
 
+  std::vector<float> runtimes;
   for (auto bench_iter = 0; bench_iter < FLAGS_benchmark_iter; ++bench_iter) {
```
More things here could be said about clearing the cache etc.

But, btw, why don't we use gbenchmark for all of this?

ilia-cher replied:
I might change to gbenchmark in the follow ups

cpuhrsch reviewed (approved):
stamped

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head

Commit:
Update on "Update intra_inter_benchmark"

Update intra_inter_benchmark

Summary:
Adding extra options into intra_inter_benchmark and minor updates

Test Plan:
PARALLEL_BACKEND=NATIVE_TBB USE_OPENMP=0 USE_TBB=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1
python setup.py develop --cmake

./build/bin/intra_inter_benchmark

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 22051 gh/ilia-cher/31/head
